pkg/operator/ceph/cluster/watcher.go

```diff
@@ -67,7 +72,8 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		return false
 	}
 
-	if !cluster.Spec.Storage.UseAllNodes {
+	if !cluster.Spec.Storage.UseAllNodes && len(cluster.Spec.Storage.Nodes) == 0 && len(cluster.Spec.Storage.VolumeSources) == 0 && len(cluster.Spec.Storage.StorageClassDeviceSets) == 0 {
+		nodesCheckedForReconcile.Add(node.Name) // adding node to nodesCheckedForReconcile Set
```

Comment:
No need for this comment as its obvious from the code what we are trying to do. Also, prefer writing comments before the code rather than on the same line. 
```suggestion
		nodesCheckedForReconcile.Add(node.Name)
```

Reply:
The condition is long., pease put that condition inside a small function and add unit tests to it.